firebaseai/ImagenScreen/ImagenViewModel.swift

```diff
@@ -37,7 +38,11 @@ class ImagenViewModel: ObservableObject {
 
   private var generateImagesTask: Task<Void, Never>?
 
-  init(firebaseService: FirebaseAI) {
+  private var sample: Sample?
+
+  init(firebaseService: FirebaseAI, sampleId: UUID? = nil) {
```

Comment:
Same here - I think it might be more straightforward to pass the sample as a reference.